django/forms/models.py

```diff
@@ -1164,10 +1164,7 @@ def add_fields(self, form, index):
         # If we're adding a new object, ignore a parent's auto-generated key
         # as it will be regenerated on the save request.
         if self.instance._state.adding:
-            if kwargs.get("to_field") is not None:
-                to_field = self.instance._meta.get_field(kwargs["to_field"])
-            else:
-                to_field = self.instance._meta.pk
+            to_field = self.instance._meta.pk
```

Comment:
Hi @sainipray — (Despite the tests) This looks suspicious to me. I want to have a slightly longer look at why we're ignoring the `to_field` here. 🤔

Reply:
Hi @carltongibson
When Child object Foreign key field uses the Parent object auto generated alternative key, so to_field have that Parent key and It's reset value None. that causes an issue IntegrityError on InlineFormSet save_model of Parent object.

I reviewed this ticket https://code.djangoproject.com/ticket/24958 and commit that just added for to_field for reset value only.
let me know if anything needs to update here.

Reply:
@sainipray — likely you're right 🙂 I'm just playing with various combinations to be sure. Thanks!